geps/gep-1897.md

```diff
@@ -125,7 +141,326 @@ This GEP is the outcome of the TLS use cases #4 and #5 in
+TLS connection must fail.  If no TLS versions are specified, then an implementation may choose to apply a self-determined
+default version.
+
+Thus, the following additions would be made to the Gateway API:
+
+```go
+// TLSConnectionPolicy provides a way to publish TLS configuration
+// that enables a gateway client to connect to a backend pod.
+type TLSConnectionPolicy struct {
+    metav1.TypeMeta   `json:",inline"`
+    metav1.ObjectMeta `json:"metadata,omitempty"`
+
+    // Spec defines the desired state of TLSConnectionPolicy.
+    Spec TLSConnectionPolicySpec `json:"spec"`
+
+    // Status defines the current state of TLSConnectionPolicy.
+    Status TLSConnectionPolicyStatus `json:"status,omitempty"`
+}
+
+// TLSConnectionPolicySpec defines the desired state of
+// TLSConnectionPolicy.
+// Note: there is no Override or Default policy configuration.
+//
+// Support: Core
+type TLSConnectionPolicySpec struct {
+    // TargetRef identifies an API object to apply policy to.
+    // Services are the only valid API target references.
+    // Note that this config applies to the entire referenced resource
+    // by default, but this default may change in the future to provide
+    // a more granular application of the policy.
+    TargetRef gatewayv1a2.PolicyTargetReference `json:"targetRef"`
+
+    // TLS contains TLS connection policy configuration.
+    TLS *TLSConnectionPolicyConfig `json:”tls”`
+}
+
+// TLSConnectionPolicyConfig contains TLS connection policy configuration.
+type TLSConnectionPolicyConfig struct {
+    // TrustedCACertRefs contains one or more references to
+    // Kubernetes objects that contain PEM-encoded TLS certificates,
+    // which are used to establish a TLS handshake between the gateway
+    // and backend pod.
+    //
+    // If the TrustedCACertRefs is empty, then the system trusted
+    // certificates should be used. If there are none, or the
+    // implementation doesn't define system trusted certificates,
+    // then a TLS connection must fail.
+    //
+    // A single TrustedCACertRef to a Kubernetes ConfigMap has "Core"
+    // support.  Implementations MAY choose to support attaching
+    // multiple certificates to a backend, but this behavior is
+    // implementation-specific.
+    //
+    // References to a resource in a different namespace are 
+    // invalid.
+    // 
+    // Support: Core - An optional single reference to a Kubernetes
+    // ConfigMapName.
+    //
+    // Support: Implementation-specific (No reference, more than one
+    // reference, or other resource types; service mesh may ignore.)
+    //
+    // +kubebuilder:validation:MaxItems=64
+    // +optional
+    TrustedCACertRefs []ConfigMapName `json:”trustedCACertRefs,omitempty”`
+
+    // AllowedSANs are the Subject Alternative Names that are allowed for
+    // this TLS connection.  It represents the hostnames that may
```

Comment:
or URL SANs ( not only hostnames SANs - spiffee uses URLs instead )

Reply:
I'm not aware of SPIFFE in any other contexts for Gateway API, so it would be great if you could open an issue on this to gauge interest instead of bringing it up here.

From what I understand, SPIFFE would want to strip the hostname from the url anyway so it could be used to validate the X.509 certificate:
https://www.envoyproxy.io/docs/envoy/latest/api-v3/extensions/transport_sockets/tls/v3/tls_spiffe_validator_config.proto